Mathlib/MeasureTheory/Group/Arithmetic.lean

```diff
@@ -516,19 +525,56 @@ instance measurableSMul₂_of_mul (M : Type*) [Mul M] [MeasurableSpace M] [Measu
 
 @[to_additive]
 instance Submonoid.measurableSMul {M α} [MeasurableSpace M] [MeasurableSpace α] [Monoid M]
-    [MulAction M α] [MeasurableSMul M α] (s : Submonoid M) : MeasurableSMul s α :=
-  ⟨fun c => by simpa only using measurable_const_smul (c : M), fun x =>
-    (measurable_smul_const x : Measurable fun c : M => c • x).comp measurable_subtype_coe⟩
+    [MulAction M α] [MeasurableSMul M α] (s : Submonoid M) : MeasurableSMul s α where
+  measurable_const_smul c := by simpa only using measurable_const_smul (c : M)
+  measurable_smul_const x := (measurable_smul_const (M := M) x).comp measurable_subtype_coe
 
 @[to_additive]
 instance Subgroup.measurableSMul {G α} [MeasurableSpace G] [MeasurableSpace α] [Group G]
     [MulAction G α] [MeasurableSMul G α] (s : Subgroup G) : MeasurableSMul s α :=
   s.toSubmonoid.measurableSMul
 
 section SMul
+variable {M X α β : Type*} [MeasurableSpace X] [SMul M X]
+  {m : MeasurableSpace α} {mβ : MeasurableSpace β} {μ : Measure α} {f : α → M} {g : α → X}
 
-variable {M X α β : Type*} [MeasurableSpace M] [MeasurableSpace X] [SMul M X]
-  {m : MeasurableSpace α} {mβ : MeasurableSpace β} {f : α → M} {g : α → X}
+section MeasurableConstSMul
+variable [MeasurableConstSMul M X]
+
+@[to_additive (attr := fun_prop, measurability)]
+lemma Measurable.const_smul (hg : Measurable g) (c : M) : Measurable (c • g) :=
+  (measurable_const_smul c).comp hg
+
+/-- Compositional version of `Measurable.const_smul` for use by `fun_prop`. -/
+@[to_additive (attr := fun_prop)
+"Compositional version of `Measurable.const_vadd` for use by `fun_prop`."]
+lemma Measurable.const_smul' {g : α → β → X} {h : α → β} (hg : Measurable ↿g) (hh : Measurable h)
+    (c : M) : Measurable fun a ↦ (c • g a) (h a) :=
+  (hg.comp <| measurable_id.prodMk hh).const_smul _
+
+@[to_additive (attr := fun_prop, measurability)]
+lemma AEMeasurable.const_smul' (hg : AEMeasurable g μ) (c : M) : AEMeasurable (c • g ·) μ :=
```

Comment:
I think that per the naming convention this is now
```suggestion
lemma AEMeasurable.fun_const_smul (hg : AEMeasurable g μ) (c : M) : AEMeasurable (c • g ·) μ :=
```
even though you didn't write the `fun` explicitely.

Reply:
Yes, I'm sure: let's fix what we can fix!